geospaas_harvesting/verify.py

```diff
@@ -0,0 +1,40 @@
+""" Verification module. """
+import os
+import sys
+from datetime import datetime
+
+import django
+import requests
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_harvesting.settings')
+django.setup()
+
+from geospaas.catalog.models import DatasetURI
+
+def main(filename):
+    """ Verifies the datasets based on their dataseturi. If the download link does not provide a
+    download availability and returns a text or html response, then the dataset uri is removed. If
+    there is no other "dataseturi" remains for the dataset, then the dataset is also removed in
+    order to be harvested again in the future with a correct and healthy "dataseturi".
+    Since the number of datasets in the database might be enormous, the datasets are retrieved
+    into with a variable named retrieved_dataset_uris with an specific length number
+    for memory management. """
+    init_index = 0
+    interval = 1000
+    with open(filename+".txt", 'w') as f:
+        while init_index < DatasetURI.objects.count():
+            try:
+                retrieved_dataset_uris = DatasetURI.objects.all()[init_index:init_index+interval]
+            except IndexError:
+                retrieved_dataset_uris = DatasetURI.objects.all()[init_index:]
+            init_index += interval
```

Comment:
This is still more complicated that necessary.
https://docs.djangoproject.com/en/3.1/ref/models/querysets/#iterator